presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OriginalFilesUtils.java

```diff
@@ -0,0 +1,119 @@
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.io.AcidUtils;
+
+import java.io.IOException;
+import java.util.List;
+
+import static io.prestosql.plugin.hive.HiveErrorCode.HIVE_CANNOT_OPEN_SPLIT;
+import static io.prestosql.plugin.hive.OriginalFileLocations.OriginalFileInfo;
+
+public class OriginalFilesUtils
+{
+    private OriginalFilesUtils()
+    {
+    }
+
+    /**
+     * Utility method to read the ORC footer and return number of rows present in the file.
+     */
+    private static Long getRowsInFile(Path splitPath,
+            HdfsEnvironment hdfsEnvironment,
+            String sessionUser,
+            OrcReaderOptions options,
+            Configuration configuration,
+            FileFormatDataSourceStats stats,
+            long fileSize)
+    {
+        OrcDataSource orcDataSource;
+        try {
+            FileSystem fileSystem = hdfsEnvironment.getFileSystem(sessionUser, splitPath, configuration);
+            FSDataInputStream inputStream = hdfsEnvironment.doAs(sessionUser, () -> fileSystem.open(splitPath));
+            orcDataSource = new HdfsOrcDataSource(
+                    new OrcDataSourceId(splitPath.toString()),
+                    fileSize,
+                    options,
+                    inputStream,
+                    stats);
+        }
+        catch (Exception e) {
+            throw new PrestoException(HIVE_CANNOT_OPEN_SPLIT, "Could not create ORC data source for file: " + splitPath.getName(), e);
+        }
+
+        OrcReader reader;
+        try {
+            reader = new OrcReader(orcDataSource, options);
+        }
+        catch (IOException e) {
+            throw new PrestoException(HIVE_CANNOT_OPEN_SPLIT, "Could not create ORC reader for file: " + splitPath.getName(), e);
+        }
+        return reader.getFooter().getNumberOfRows();
+    }
+
+    /**
+     * Returns total number of rows present before the given original file in the same bucket.
+     * example: if bucket-1 has original files
+     *  000000_0 -> X0 rows
+     *  000000_0_copy1 -> X1 rows
+     *  000000_0_copy2 -> X2 rows
+     *
+     * for 000000_0_copy2, it returns (X0+X1)
+     */
+    public static long getRowCount(List<OriginalFileInfo> originalFileInfos,
+            Path splitPath,
+            HdfsEnvironment hdfsEnvironment,
+            String sessionUser,
+            OrcReaderOptions options,
+            Configuration configuration,
+            FileFormatDataSourceStats stats)
+    {
+        long rowCount = 0;
+        for (OriginalFileInfo originalFileInfo : originalFileInfos) {
+            Path path = new Path(splitPath.getParent() + "/" + originalFileInfo.getName());
+            try {
+                // Check if the file belongs to the same bucket and comes before 'reqPath' in lexicographic order.
+                if (isSameBucket(splitPath, path, configuration) && path.compareTo(splitPath) < 0) {
```

Comment:
since we BHSL divides information per bucket, do we need `isSameBucket` here?

Reply:
yeah, it is redundant. removed.